docs/technical_documents/planning_component.md

```diff
@@ -0,0 +1,33 @@
+# PnC Component Introduction
+
+_**Tip**: to read the equations in the document, you are recommended to use Chrome with [a plugin](https://chrome.google.com/webstore/detail/tex-all-the-things/cbimabofgmfdkicghcadidpemeenbffn) or copy the latex equation to [an online editor](http://www.hostmath.com/)_
+
+# Introduction
+
+This article describes the architecture of planning module. It mainly consists of the following parts:
+1. Cyber component related knowledge
+2. How the messages are received and sent
+3. How the trajectory is arranged and made up
+
+# Where is the code
+
+Please refer to [code](https://github.com/ApolloAuto/apollo/blob/master/modules/planning/planning_component.h).
+
+# Code Reading
+
+## Cyber component related knowledge
+Please refer to [Cyber](https://github.com/ApolloAuto/apollo/blob/master/docs/cyber/CyberRT_Terms.md).
+
+## How the messages are received and sent
+![Diagram](images/planning_component_api.png)
+
+1. Input
+Messages are partially input from readers, which are defined in PlanningComponent::Init(). For Planning component, readers receive message types of RoutingResponse/TrafficLightDetection/PadMessage/Stories/MapMsg and store them in local for processing afterward.
+Another part of input messages are defined in PlanningComponent::Proc(), where the planning process is called when new messages arrives.
+2. Process. PlanningComponent::Proc() function works like Reader's callback function. Each time there are new messages of prediction::PredictionObstacles/canbus::Chassis/localization::LocalizationEstimate, this process is implemented till the end of this function.
+3. Output. PlanningComponent output the message by the following line `planning_writer_->Write(adc_trajectory_pb)`, in which the writer send the final adc_trajectory_pb message to the cyber system. The writer is initialized in PlanningComponent::Init(), just after the readers are initialized.
+
+## How the trajectory is arranged and made up
```

Comment:
"How the trajectory as planning component output is produced" ?

Reply:
Changed to "How the trajectory as planning component output is produced"